adapters/huaweiads/huaweiads.go

```diff
@@ -559,7 +559,10 @@ func getNativeFormat(adslot30 *adslot30, openRTBImp *openrtb2.Imp) error {
 		// Only one of the {title,img,video,data} objects should be present in each object.
 		if asset.Video != nil {
 			numVideo++
-			continue
+			formats = popularSizes
+			if (asset.Video.W != 0 && asset.Video.H != 0) && !isSizeInArray(popularSizes, asset.Video.W, asset.Video.H) {
+				formats = append(formats, format{asset.Video.W, asset.Video.H})
+			}
```

Comment:
`isSizeInArray(popularSizes, asset.Video.W, asset.Video.H) ` is 0(N) look up. Could use map for 0(1) look up